builtin/credential/jwt/path_config.go

```diff
@@ -163,6 +167,91 @@ func (b *jwtAuthBackend) config(ctx context.Context, s logical.Storage) (*jwtCon
 	return config, nil
 }
 
+func contactIssuer(ctx context.Context, uri string, data *url.Values, ignoreBad bool) ([]byte, error) {
+	var req *http.Request
+	var err error
+	if data == nil {
+		req, err = http.NewRequest("GET", uri, nil)
+	} else {
+		req, err = http.NewRequest("POST", uri, strings.NewReader(data.Encode()))
+	}
+	if err != nil {
+		return nil, nil
+	}
+	if data != nil {
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	}
+
+	client, ok := ctx.Value(oauth2.HTTPClient).(*http.Client)
+	if !ok {
+		client = http.DefaultClient
+	}
+	resp, err := client.Do(req.WithContext(ctx))
+	if err != nil {
+		return nil, nil
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil
+	}
+
+	if resp.StatusCode != http.StatusOK && (!ignoreBad || resp.StatusCode != http.StatusBadRequest) {
+		return nil, fmt.Errorf("%s: %s", resp.Status, body)
+	}
+
+	return body, nil
+}
+
+// Discover the device_authorization_endpoint URL and store it in the config
+// This should be in coreos/go-oidc but they don't yet support device flow
```

Comment:
Does https://github.com/coreos/go-oidc/pull/365 do what you want? I'm not familiar enough with the ecosystem to know, but my quick read is that it does. Perhaps we just need to bump the `go-oidc` version in OpenBao?

Reply:
That does indeed look promising, a feature added pretty recently. I will look into it.

Reply:
So go-oidc is used only from https://github.com/hashicorp/cap.  The latest tagged version of that does use a new enough version of go-oidc, but it does not expose the DeviceAuthURL information to the caller of that library.  I'm trying to test to see if the addition of an `Endpoint()` function that calls the go-oidc [`Endpoint()`](https://github.com/coreos/go-oidc/blob/v3/oidc/oidc.go#L282) function will be sufficient, and if so I plan to try to submit that as a PR.  Perhaps openbao should be forking that module as well.

Reply:
That works, so I made hashicorp/cap#139

Reply:
Thanks @DrDaveD! If they don't merge it, we can think about forking. But I'd like to give them the chance to keep the dependencies the same :-)

Reply:
Meanwhile how about if we use the current implementation and optimize it later after cap is updated and has a new tag?

Reply:
@DrDaveD We can pull the updated cap without waiting for a new tag. Let's give it a bit, while it'd be nice to merge this, I tagged it for 2.1.0 and we'll still need to merge transactional storage before 2.1.0 will be release ready...

Reply:
hashicorp/cap#139 has been replaced by hashicorp/cap#140 after feedback from the developers.